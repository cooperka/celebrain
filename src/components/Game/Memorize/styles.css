.Memorize img {
  height: 30em;
}
