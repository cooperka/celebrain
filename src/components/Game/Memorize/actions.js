import { setGameState } from '../actions';
import { gameState } from '../reducers';

// --- Action types

export const actionTypes = {
  MEMORIZE_NEXT: 'MEMORIZE_NEXT',
};

// --- Action creators

export function next() {
  return (dispatch, getState) => {
    const { memorize: { currIndex, numItems } } = getState();
    const nextIndex = currIndex + 1;

    if (nextIndex >= numItems) {
      dispatch(setGameState(gameState.RECALL));
    }

    dispatch({ type: actionTypes.MEMORIZE_NEXT });
  };
}
