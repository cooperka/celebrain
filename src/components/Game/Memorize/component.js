// @flow

import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Button } from 'material-ui';

import { next } from './actions';

import './styles.css';

type Props = {
  classes: any,
  imageData: any,

  imageOrder: any,
  currIndex: number,
  handleNext: () => void,
};

class Memorize extends Component<Props> {

  static mapStateToProps(state) {
    return {
      imageOrder: state.game.imageOrder,
      currIndex: state.memorize.currIndex,
    };
  }

  static mapDispatchToProps(dispatch) {
    return {
      handleNext: () => {
        dispatch(next());
      },
    };
  }

  render() {
    const { classes, imageData, imageOrder, currIndex, handleNext } = this.props;
    const numItems = imageOrder.length;

    if (currIndex >= numItems) return null;

    const currCeleb = imageData[imageOrder[currIndex]];

    return (
      <div className="Memorize">
        <div className="title">Memorize</div>
        <div className="section">
          <img className={classes.shadowed} src={currCeleb.image} alt="Celebrity" />
          <div className="subtitle">
            <a href={`https://en.wikipedia.org/?curid=${currCeleb.id}`}>{currCeleb.name}</a>
          </div>
        </div>
        <div className="section">
          <Button onClick={handleNext}>Next {`(${currIndex + 1}/${numItems})`}</Button>
        </div>
      </div>
    );
  }

}

export default connect(Memorize.mapStateToProps, Memorize.mapDispatchToProps)(Memorize);
