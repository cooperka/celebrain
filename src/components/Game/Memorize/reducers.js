import { actionTypes } from './actions';

const initialState = {
  currIndex: 0,
  numItems: 1,
};

export function memorizeReducer(state = initialState, action = {}) {
  switch (action.type) {
    case actionTypes.MEMORIZE_NEXT:
      return {
        ...state,
        currIndex: state.currIndex + 1,
      };

    default:
      return state;
  }
}
