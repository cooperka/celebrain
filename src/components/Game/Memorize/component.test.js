// @flow

import React from 'react';
import renderer from 'react-test-renderer';

import Memorize from './component';

import imageData from '../../../../public/celebs.json';

it('renders', () => {
  const tree = renderer.create(
    <Memorize imageData={imageData} currIndex={0} numItems={1} imageOrder={[0]} />,
  );
  expect(tree.toJSON()).toMatchSnapshot();
});
