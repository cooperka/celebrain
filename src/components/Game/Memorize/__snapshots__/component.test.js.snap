// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="Memorize"
>
  <div
    className="title"
  >
    Memorize
  </div>
  <div
    className="section"
  >
    <a
      href="https://commons.wikimedia.org/wiki/File:Arnold_Schwarzenegger_in_Film_Festival.jpg"
    >
      <img
        alt="Celebrity"
        className="CelebImage-shadowed-1"
        src="https://upload.wikimedia.org/wikipedia/commons/thumb/1/1c/Arnold_Schwarzenegger_in_Film_Festival.jpg/800px-Arnold_Schwarzenegger_in_Film_Festival.jpg"
      />
    </a>
    <div
      className="subtitle"
    >
      <a
        className="name"
        href="https://en.wikipedia.org/?curid=1806"
      >
        Arnold Schwarzenegger
      </a>
    </div>
  </div>
  <div
    className="section"
  >
    <button
      className="MuiButtonBase-root-17 MuiButton-root-2"
      disabled={false}
      onBlur={[Function]}
      onClick={undefined}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label-3"
      >
        Next 
        (1/1)
      </span>
      <span
        className="MuiTouchRipple-root-19"
      />
    </button>
  </div>
</div>
`;
